pennylane/templates/state_preparations/basis_qutrit.py

```diff
@@ -112,7 +113,10 @@ def compute_decomposition(basis_state, wires):  # pylint: disable=arguments-diff
         """
 
         op_list = []
+        tshift = qml.math.array([[0, 0, 1], [1, 0, 0], [0, 1, 0]])
         for wire, state in zip(wires, basis_state):
-            for _ in range(0, state):
-                op_list.append(qml.TShift(wire))
+            mat = qml.math.matrix_power(tshift, state)
+            op = qml.ops.QutritUnitary(mat, wires=wire)
+            op_list.append(op)
+
```

Comment:
to avoid QubitUnitary (since it is not hardware compatible), you could use this:

`qml.pow(qml.TShift(wires = wire),state)`

qml.pow returns an operator, so the idea is the same without access the matrix. 

Note that if this works, you may not need to add matrix power in the `dispach.py` file

Reply:
The issue is that the `pow` method for `qml.TShift` is not jittable. It uses the `pow` method from `Operator` which has control flow to check if the exponent is `0` or `1`.